libyear-gradle-plugin/src/main/kotlin/com/libyear/traversal/DependencyTraversal.kt

```diff
@@ -27,7 +28,13 @@ class DependencyTraversal private constructor(
         if (maxTransitiveDepth != null && depth > maxTransitiveDepth) {
           continue
         }
+
+        if (excludedPackages.any { selected.moduleVersion.toString().contains(it, ignoreCase = true) }) {
+          continue
```

Comment:
In any case, please add a log message which pattern matched which dependency on INFO log level. Build authors must be able to diagnose the exclusion patterns by running Gradle on info log level, as in `gradle --info`, to see why a dependency might not show up in their report.
The approximate shape of the message is "Excluded dependency 'org.sl4j:slf4j' because pattern 'org.slf4j:*' matched".

A cherry on top consisted of printing the number of excluded dependencies in the console on each run - if that number is very high, the report is going to be extremely underestimating, or the build author has put in a too broad exclusion pattern by accident.